lbry/extras/daemon/daemon.py

```diff
@@ -2495,19 +2540,34 @@ async def jsonrpc_claim_search(self, **kwargs):
 
         Returns: {Paginated[Output]}
         """
-        wallet = self.wallet_manager.get_wallet_or_default(kwargs.pop('wallet_id', None))
-        if "claim_ids" in kwargs and not kwargs["claim_ids"]:
-            kwargs.pop("claim_ids")
-        if {'claim_id', 'claim_ids'}.issubset(kwargs):
-            # TODO: use error from lbry.error
-            raise ValueError("Only 'claim_id' or 'claim_ids' is allowed, not both.")
-        if kwargs.pop('valid_channel_signature', False):
-            kwargs['signature_valid'] = 1
-        if kwargs.pop('invalid_channel_signature', False):
-            kwargs['signature_valid'] = 0
-        if 'has_no_source' in kwargs:
-            kwargs['has_source'] = not kwargs.pop('has_no_source')
+        if self.ledger.config.get('use_go_hub'):
+            host = self.ledger.network.client.server[0]
+            port = "50051"
+            kwargs['new_sdk_server'] = f"{host}:{port}"
+            if kwargs.get("channel"):
+                channel = kwargs.pop("channel")
+                channel_obj = (await self.jsonrpc_resolve(channel))[channel]
+                if isinstance(channel_obj, dict):
+                    # This happens when the channel doesn't exist
+                    kwargs["channel_id"] = ""
+                else:
+                    kwargs["channel_id"] = channel_obj.claim_id
+            kwargs = fix_kwargs_for_hub(**kwargs)
+        else:
+            # Don't do this if using the hub server, it screws everything up
+            if "claim_ids" in kwargs and not kwargs["claim_ids"]:
+                kwargs.pop("claim_ids")
+            if {'claim_id', 'claim_ids'}.issubset(kwargs):
+                # TODO: use error from lbry.error
+                raise ValueError("Only 'claim_id' or 'claim_ids' is allowed, not both.")
```

Comment:
this should be a lbry.error, not a ValueError

Reply:
Isn't it a `ValueError` though? It is an error and it's because the wrong value was passed in.

Reply:
I originally preserved what is [done in master](https://github.com/lbryio/lbry-sdk/blob/master/lbry/extras/daemon/daemon.py#L2502), but since there was a TODO I added it quickly as `ConflictingInputValueError`